## Objective and methods

An automated framework for assessing, appraising, comparing and contrasting existing pipelines and the analysis outputs they provide for input data as generated by researchers as well as the exemplar datasets.

We are running multiple pipelines to discover variants without being able to appraise the accuracy of the variant calls of each of the tools. Very often set operations on the outputs yield unexpected results. Scientists, rightly, have concern over the validity of such outputs and rely heavily on consensus from the multiple tools.


### Outline of methods (1 page)

Multiple genomic read aligners, assemblers, and variant callers will be assessed on their performance using many sets of simulated read data. The data will be obtained by permutation from a “known-good” source, such as one of the human chromosomes. The controls will be sets of “perfect reads” generated as an idealization of the sequencing process, for the specified type, with a specific allele frequency. The idealized reads will then be permuted in a fashion that mimics both the introduction of sequencing errors as well as the introduction of sequencing biases and the incorporation of a range of allele frequencies. Each tool will be run on the sample data sets, and then the results evaluated using the appropriate standard criteria, including the ability to handle polyploid data correctly and efficiently.

Development external, using external system to facilitate transparency and accessibility of work with collaborators.

Development of Cloud capability.

#### Use cases

We have three use cases to ensure we have the volume and variety of data to trail and stress the framework and ensure practical utility.

**David Chagne ??**

**Samantha Baldwin or Maren Wellenreuther ??**

**Rob Elshire or International ??**

### Data

For the project we will use existing data that has been collected, or is planned to be, by the named researchers in the use cases.

### IP and MOU

No known IP restrictions?

MOU with Elshire Group. -- Do we want to have some standard thing with NZOSS as the holder of the BBS project ip??

### Risks
