## Vision and implementation - beyond this project 
**(Maximum 2 pages)**
1. _What is your vision/big picture for this work, and the timeframe for realisation of the benefits or impacts of the vision/big picture?_
2. _Who are the likely end-users and how could they benefit? Quantify benefits if possible._
3. _What are the key steps in the implementation pathway when this project is completed?  Note any critical factors in achieving uptake and expected impact, including funding, accessing new or different capabilities, and management of IP and FTO. Indicate the likely timeline for the key steps leading to the benefits or impacts.  You may provide more detail on the next steps than later ones_

The vision of this project is to develop a robust system that will accelerate the in-silico component of genome-informed breeding processes, i.e. computing variant information from sequencing data for implementing genomic selection in a range of PFR breeding programmes (pipfruit, kiwifruit, finfish, berryfruit, arable crops). This acceleration will be provided by sole automation of the necessary computing pipelines as well as an enhanced accuracy of variant "calls". A complete automated pipeline cannot be developed in a 2-year timefame as a discovery science project provides as the current variant calling pipelines do not provide the necessary precision. The sole multitude of pipelines for variant calling as well as variability of their outputs will make it necessary to create a framework for comparative analysis of these pipelines before any comprehensive solution can be considered. However, such a comparative tool can be developed within the 2-year framework.

The targeted end-users are primarily the PFR plant and finfish breeding community, as well as other research applications that requires variant information. 
- Rapidly acquired variant information is crucial for implementing genomic selection on fast growing breeding seedlings and hatchlings. Breeders decision often need to be taken in a defined window, e.g. for seedlings before the young trees would be transferred to new pots or orchard. 
- Accurate variant information is of critical importance for reliable estimation of breeding values during genomic selection. False positive variants for instance can misestimate breeding values calculated from genomic prediction models.

Future work will have to comprise the optimisation of variant calling pipelines. 

During the programme we will: [THIS IS MNOT THE RIGHT SECTION FOR THIS, FOCUS ON BEYOND THE PROJECT HERE]

- Produce an open source trusted reproducible research framework for calling variants from GBS and resequencing data sets.
- Automation of variant calling pipelines with trusted results
- Plug and play usability for end users
- A usable framework with a clear path for continuous improvement.
- Establishing the scientific value of developing an interpretive context.
